ui/views.py

```diff
@@ -123,6 +123,58 @@ def create_repo(request):
     )
 
 
+def get_vocabularies(facets):
+    """
+    Parse facet information for the template.
+    It will return a dictionary that looks like this:
+
+    {
+        (u'13', u'difficulty'): [(38, u'medium', 23), (17, u'hard', 19)],
+        (u'15', u'prerequisite'): [(44, u'yes', 23)]
+    }
+
+    The keys are tuples with Vocabulary ID and name for the key,
+    and a list of tuples containing id, label, and count for the terms.
+
+    This is for ease-of-use in the template, where the integer primary keys
+    are used for search links and the names/labels are used for display.
+
+    Args:
+        facets (dict): facet ID with terms & term counts
+    Returns:
+        vocabularies (dict): dict of vocab info to term info
+    """
+
+    if "fields" not in facets:
+        return {}
+    vocab_ids = []
+    term_ids = []
+    for vocab_id, counts in facets["fields"].items():
+        if not vocab_id.isdigit():
+            continue
+        vocab_ids.append(int(vocab_id))
+        for term_id, count in counts:
+            term_ids.append(term_id)
+
+    vocabs = {
+        x: y for x, y in Vocabulary.objects.filter(
+            id__in=vocab_ids).values_list('id', 'name')
+    }
+    terms = {
+        x: y for x, y in Term.objects.filter(
+            id__in=term_ids).values_list('id', 'label')
+    }
+    vocabularies = {}
+    for v_id, t_data in facets["fields"].items():
```

Comment:
I won't argue further on changing the facet key, but please change v_id to something more descriptive. If it can contain `resource_type` or `run` it shouldn't be named `v_id`